### PR TITLE
[#160888842] Changed button action retries for invalid_amount error

### PR DESCRIPTION
**Pull request by laurapugliese**
If the error during transaction is INVALID_AMOUNT and the user has manually entered the data of notice go back to the screen to allow the user to modify the data

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#160888842](https://www.pivotaltracker.com/story/show/160888842): Se inserisco manualmente i dati dell'avviso di pagamento e ottengo un errore INVALID_AMOUNT voglio poter aggiornare l'ammontare senza dover reinserire tutti i dati dell'avviso. 

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by laurapugliese** [comment]
This PR aims to handle the INVALID_AMOUNT error during the transaction. If the user comes from the manual entry of the payment notice on pressing the 'Retry' button in the error screen, it is taken to the previous screen and the 'Amount' field is reset.

invalid_amount_retry

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx [hunk 1]
```diff
@@ -208,7 +211,12 @@ class TransactionErrorScreen extends React.Component<Props> {
 }
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  navigateToWalletHome: () => dispatch(navigateToWalletHome())
+  navigateToWalletHome: () => dispatch(navigateToWalletHome()),
+  navigateToPaymentManualDataInsertion: (isInvalidAmount: boolean) =>
```
Comment (Undermaken) [inline ts/screens/wallet/payment/TransactionErrorScreen.tsx]: Does it seem to be unused?

Comment (laurapugliese) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx]: Yes :) removed (c3a8dd2)

File: ts/screens/wallet/payment/ScanQrCodeScreen.tsx [hunk 2]
```diff
@@ -292,7 +292,9 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   navigateToWalletHome: () => dispatch(navigateToWalletHome()),
   navigateToPaymentManualDataInsertion: () =>
-    dispatch(navigateToPaymentManualDataInsertion()),
+    dispatch(
+      navigateToPaymentManualDataInsertion({ isInvalidAmount: undefined })
```
Comment (fpersico) [inline ts/screens/wallet/payment/ScanQrCodeScreen.tsx]: Are you sure `isInvalidAmount: undefined` is needed?

Comment (laurapugliese) [reply ts/screens/wallet/payment/ScanQrCodeScreen.tsx]: @fpersico I have made "params" optional in navigation.ts and now it is no longer necessary. (b28d7f1)

File: ts/screens/wallet/payment/ManualDataInsertionScreen.tsx [hunk 3]
```diff
@@ -236,17 +254,21 @@ class ManualDataInsertionScreen extends React.Component<Props, State> {
...
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({
+const mapDispatchToProps = (dispatch: Dispatch, props: OwnProps) => ({
   navigateToWalletHome: () => dispatch(navigateToWalletHome()),
   navigateToTransactionSummary: (
     rptId: RptId,
     initialAmount: AmountInEuroCents
   ) => {
+    if (props.navigation.getParam("isInvalidAmount")) {
+      props.navigation.setParams({ isInvalidAmount: false });
```
Comment (fpersico) [inline ts/screens/wallet/payment/ManualDataInsertionScreen.tsx]: Why you need to set the param here?

Comment (fpersico) [reply ts/screens/wallet/payment/ManualDataInsertionScreen.tsx]: mapStateToProps must be used to get stuff from state and create props from the stuff.

Comment (laurapugliese) [reply ts/screens/wallet/payment/ManualDataInsertionScreen.tsx]: Fixed in the commit (9ca441a). I have used the willFocus event.

Comment (laurapugliese) [reply ts/screens/wallet/payment/ManualDataInsertionScreen.tsx]: @fpersico I did another test on a physical android device after the change (screencast below). I confirm that the amount is reset before the screen is visible.

retry_nexus

**Pull request merged**